src/network/networkd-network.h

```diff
@@ -68,6 +68,14 @@ typedef enum LLDPMode {
         _LLDP_MODE_INVALID = -1,
 } LLDPMode;
 
+typedef struct DUID {
+        /* Value of Type in [DHCP] section */
+        DUIDType type;
+
+        uint8_t duid_len;
+        uint8_t duid[MAX_DUID_LEN];
+} DUID;
```

Comment:
I think structs called "foo" should not have a field called "foo"... Also, let's not abbreviate unnecessarily in types. Maybe turn this into:

```
typedef struct {
        DUIDType type;
        sizet length;
        uint8_t data[MAX_DUID_LENGTH];
} DUID;
```